packages/@aws-cdk/pipelines/lib/blueprint/blueprint.ts

```diff
@@ -0,0 +1,93 @@
+import * as cdk from '@aws-cdk/core';
+import { Step } from './step';
+import { AddStageOpts, Wave } from './wave';
+
+/**
+ * Construction properties for a `Blueprint`.
+ */
+export interface BlueprintProps {
+  /**
+   * The build step that produces the CDK Cloud Assembly
+   *
+   * The primary output of this step needs to be the `cdk.out` directory
+   * generated by the `cdk synth` command.
+   *
+   * If you use `SynthStep`, the output directory will automatically be
+   * configured correctly; if you use a different kind of `Step`, you
+   * may need to configure it explicitly.
+   */
+  readonly synthStep: Step;
```

Comment:
I would prefer to do it later. In any case, today this is still possible because you would do the following:

```ts
new CodePipelinePipeline({
  synth: CodePipelineSource.gitHub('my/repo'),
});
```

The only thing we need from the `synth` step is that it has a primary output, which is assumed to be the cloud assembly.